### PR TITLE
Fixed a bug in convertDayOrdinalToPositive. (Last Sunday of February wasn't working)

### PR DESCRIPTION
**Pull request by tkoop**
> _I noticed a problem we were having with our events. There was a case where an event was the last Tuesday of the month, and it happened that the last day of February (28 days, exactly 4 weeks) was the last Tuesday (or something like that; I don't remember). So I noticed that the function convertDayOrdinalToPositive was returning the wrong number in this case. So I fixed the code in that function by rewriting it. And it's a lot simpler now in my opinion._

### CODE REVIEW

**Comment by u01jmg3** [comment]
> _This includes the use of 4 spaces over tabs for indentation_

I can see tabs

> _Using single quotes (`'`) where possible_

I can see unnecessary double quotes

---

- As tabs have been used I can't see what you have changed in the diff because Git sees every line as different when indentation does not match.
- Equally 'Fixed a bug in convertDayOrdinalToPositive' gives me no context regarding the problem you found and fixed.
- It also looks like existing arrays could be utilised such as `$dayOrdinals` and `$weekdays` to prevent repeated code.

Whilst I appreciate contributions please follow the guidelines I spend my time carefully writing.

**Comment by u01jmg3** [comment]
@tkoop: going to need more info about the bug you found before I can safely merge this code.

**Comment by tkoop** [comment]
I noticed a problem we were having with our events.  There was a case where an event was the last Tuesday of the month, and it happened that the last day of February (28 days, exactly 4 weeks) was the last Tuesday (or something like that; I don't remember).  So I noticed that the function convertDayOrdinalToPositive was returning the wrong number in this case.  So I fixed the code in that function by rewriting it.  And it's a lot simpler now in my opinion.

File: src/ICal/ICal.php [hunk 1]
```diff
@@ -2393,31 +2393,27 @@ protected function numberOfDays($days, $start, $end)
...
+		$dayOrdinals = $this->dayOrdinals;
+
+		// We only care about negative BYDAY values
+		if ($dayNumber >= 1) {
+			return $dayOrdinals[$dayNumber];
+		}
+
+		$ords = array("", "second", "third", "fourth", "fifth");
+		$weekdays = array("SU"=>"Sunday", "MO"=>"Monday", "TU"=>"Tuesday", "WE"=>"Wednesday", "TH"=>"Thursday", "FR"=>"Friday", "SA"=>"Saturday");
+		$month = date("F", $timestamp);
```
Comment (u01jmg3) [inline src/ICal/ICal.php]: What happens when `$timestamp` is a DateTime object?

If you run this against `.\examples\ICal.ics`, you will get:

- `Warning: date() expects parameter 2 to be int, object given`

I would test against this file to make sure your method changes return the same output.

**Comment by u01jmg3** [comment]
Closing due to a lack of activity and my review not being addressed. Happy to re-open.

**Pull request closed without merging**